File: dotnet/Lexy/Lexy.RunTime/RunTime/BuiltInDateFunctions.cs
using System;

namespace Lexy.RunTime.RunTime
{
    public static class BuiltInDateFunctions
    {
        public static DateTime Now() => DateTime.Now;
        public static DateTime Today() => DateTime.Today;

        public static decimal Year(DateTime value) => value.Year;
        public static decimal Month(DateTime value) => value.Month;
        public static decimal Day(DateTime value) => value.Day;
        public static decimal Hour(DateTime value) => value.Hour;
        public static decimal Minute(DateTime value) => value.Minute;
        public static decimal Second(DateTime value) => value.Second;

        public static decimal Years(DateTime end, DateTime start) => DateTimeSpan.CompareDates(end, start).Years;
        public static decimal Months(DateTime end, DateTime start) => DateTimeSpan.CompareDates(end, start).Months;
        public static decimal Days(DateTime end, DateTime start) => DateTimeSpan.CompareDates(end, start).Days;
        public static decimal Hours(DateTime end, DateTime start) => DateTimeSpan.CompareDates(end, start).Hours;
        public static decimal Minutes(DateTime end, DateTime start) => DateTimeSpan.CompareDates(end, start).Minutes;
        public static decimal Seconds(DateTime end, DateTime start) => DateTimeSpan.CompareDates(end, start).Seconds;
    }
}

File: typescipt/src/runTime/builtInDateFunctions.ts


export class BuiltInDateFunctions {
   public static now(): DateTime {
     return DateTime.Now;
   }

   public static today(): DateTime {
     return DateTime.Today;
   }

   public static year(value: DateTime): decimal {
     return value.Year;
   }

   public static month(value: DateTime): decimal {
     return value.Month;
   }

   public static day(value: DateTime): decimal {
     return value.Day;
   }

   public static hour(value: DateTime): decimal {
     return value.Hour;
   }

   public static minute(value: DateTime): decimal {
     return value.Minute;
   }

   public static second(value: DateTime): decimal {
     return value.Second;
   }

   public static years(end: DateTime, start: DateTime): decimal {
     return DateTimeSpan.CompareDates(end, start).Years;
   }

   public static months(end: DateTime, start: DateTime): decimal {
     return DateTimeSpan.CompareDates(end, start).Months;
   }

   public static days(end: DateTime, start: DateTime): decimal {
     return DateTimeSpan.CompareDates(end, start).Days;
   }

   public static hours(end: DateTime, start: DateTime): decimal {
     return DateTimeSpan.CompareDates(end, start).Hours;
   }

   public static minutes(end: DateTime, start: DateTime): decimal {
     return DateTimeSpan.CompareDates(end, start).Minutes;
   }

   public static seconds(end: DateTime, start: DateTime): decimal {
     return DateTimeSpan.CompareDates(end, start).Seconds;
   }
}
